newIDE/app/src/UI/SemiControlledAutoComplete.js

```diff
@@ -141,17 +142,15 @@ const filterFunction = (
 
 const handleChange = (
   event: SyntheticKeyboardEvent<HTMLInputElement>,
-  option: Option | string,
+  option: Option,
   props: Props
 ): void => {
   if (option.type !== 'separator') {
-    if (typeof option === 'string') props.onChange(option);
+    if (option.onClick) option.onClick();
     else {
-      if (option.onClick) option.onClick();
-      else
-        props.onChoose
-          ? props.onChoose(option.value)
-          : props.onChange(option.value);
+      if (props.onChoose) props.onChoose(option.value);
+      props.onChange(option.value);
```

Comment:
Are you sure we should call onChoose AND onChange?

Reply:
No, I'm not sure about them and I forgot to ask that. I've never come across onChoose calls (I know about onClick and onChange only, and there is a comment which mentions onClick has the highest priority). Can you explain what is the purpose and priority of onChoose?

Reply:
The idea is that onChange is to be called when we update the content of the text (in a "semi controlled" fashion, i.e: only when we blur the field).

While onChoose is to be called when we *select* (i.e: click or press enter) an existing item (not something that was entered manually). This is used in ObjectGroupEditor: an object is added to the group when `onChoose` is called. `onChange` is just used to remember the name entered.